src/java.desktop/share/classes/java/awt/image/BufferedImage.java

```diff
@@ -1509,31 +1509,7 @@ public WritableRaster copyData(WritableRaster outRaster) {
      * @see #getData(Rectangle)
     */
     public void setData(Raster r) {
-        int width = r.getWidth();
-        int height = r.getHeight();
-        int startX = r.getMinX();
-        int startY = r.getMinY();
-
-        int[] tdata = null;
-
-        // Clip to the current Raster
-        Rectangle rclip = new Rectangle(startX, startY, width, height);
-        Rectangle bclip = new Rectangle(0, 0, raster.width, raster.height);
-        Rectangle intersect = rclip.intersection(bclip);
-        if (intersect.isEmpty()) {
```

Comment:
Does the "raster.setRect" has equivalent optimizations? Does it seem that method implements handcrafter "intersection"?

Reply:
I do not know why `WritableRaster.setRect(Raster)` computes intersection with its own code in the method body instead of delegating to `Rectangle.intersection(Rectangle)` in the same way as `BufferedImage.setData(Raster)` does. For now I see no technical reason. I'm tempted to add a commit to this pull request for replacing the intersection calculation of `WritableRaster.setRect(Raster)`, but abstained for now because I though that reviewers or sponsors may want to minimize the changes. However I believe that this replacement would be desirable at least for the extra robustness of `Rectangle` regarding underflow and overflow.

Reply:
Can we prove the overflow issue there by some test? If yes then probably it would be useful to fix it as well, probably even as separate bug which can be later backported.

Reply:
In theory the following code would demonstrate the overflow:

```java
var sm = new BandedSampleModel(DataBuffer.TYPE_BYTE, 100, 100, 1);
var r1 = Raster.createWritableRaster(sm, new Point(Integer.MAX_VALUE - 150, 0));
var r2 = Raster.createWritableRaster(sm, new Point(Integer.MAX_VALUE -  50, 0));    // Overflow on X.
r1.setRect(r2);
```

It practice it does not happen so easily because the `Raster` constructor detects in advance the overflow and throws a `RasterFormatException` saying _"overflow condition for X coordinates of Raster"_. However because the `Raster.minX` field is protected and non-final, the constructor check can be bypassed as below:

```java
var r2 = new WritableRaster(sm, new Point()) {
    {   // Constructor
        minX = Integer.MAX_VALUE - 50;
        sampleModelTranslateX = minX;
    }
};
```

In which case the overflow happens and manifest itself with an `ArrayIndexOutOfBoundsException`. I do not know if this is considered a too convolved case for being worth a fix. However using the `BufferedImage` code, in addition of being robust to this case, would also be simpler.